scrape_books/spiders/books.py

```diff
@@ -0,0 +1,49 @@
+import scrapy
+
+from scrapy.http import Response
+from typing import Any
+
+from scrape_books.items import ScrapeBooksItem
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs: Any) -> None:
+        for book in response.css("article.product_pod"):
+            detail_url = response.urljoin(book.css("a::attr(href)").get())
+            yield scrapy.Request(
+                url=detail_url,
+                callback=self._parse_book_detail
+            )
+
+        next_page = response.css("li.next a::attr(href)").get()
+        if next_page is not None:
+            yield response.follow(next_page, callback=self.parse)
+
+    def _parse_book_detail(self, response: Response) -> dict[str, Any]:
+        yield ScrapeBooksItem(
+            title=response.css(
+                ".product_main > h1::text"
+            ).get(),
+            price=response.css(
+                ".product_main > .price_color::text"
+            ).get(),
+            amount_in_stock=response.css(
+                ".product_main > p.instock.availability::text"
+            ).getall(),
+            rating=response.css(
+                ".product_main p.star-rating::attr(class)"
+            ).get(),
```

Comment:
The `rating` extraction assumes the class attribute contains the rating information. Ensure that this assumption holds true for all pages to prevent incorrect data extraction.